nestkernel/event.h

```diff
@@ -746,63 +746,32 @@ class SecondaryEvent : public Event
   virtual fwit& operator>>( fwit& pos ) = 0;
 };
 
-/**
- * Iterator class for the GapJEvent data array.
- * Generating a copy of the array at several times is too time consuming,
- * therefore we use iterators pointing to the "old" memory location of the array
- */
-class CoeffArrayIterator
+template < typename T >
+size_t size_uint_t( T )
 {
+  if ( sizeof( T ) < sizeof( uint_t ) )
+    return 1;
+  else if ( sizeof( T ) % sizeof( uint_t ) != 0 )
+    return sizeof( T ) / sizeof( uint_t ) + 1;
+  else
+    return sizeof( T ) / sizeof( uint_t );
+}
 
-  friend class GapJEvent;
-
-private:
-  fwit pos_;
-
-public:
-  CoeffArrayIterator()
-  {
-  }
-
-  CoeffArrayIterator( const fwit& rhs )
-    : pos_( rhs )
-  {
-  }
-
-  CoeffArrayIterator( const dfwit& rhs )
-  {
-    fwit iit( reinterpret_cast< uint_t* >( &( *rhs ) ) );
-    pos_ = iit;
-  }
-
-  CoeffArrayIterator( const CoeffArrayIterator& rhs )
-    : pos_( rhs.pos_ )
-  {
-  }
-
-  CoeffArrayIterator& operator++()
-  {
-    size_t inc = 0;
-    if ( sizeof( double ) % sizeof( uint_t ) != 0 )
-      inc = sizeof( double ) / sizeof( uint_t ) + 1;
-    else
-      inc = sizeof( double ) / sizeof( uint_t );
-
-    pos_ += inc;
-
-    return ( *this );
-  }
-
-  double& operator*()
-  {
-    return *reinterpret_cast< double* >( &( *pos_ ) );
-  }
+template < typename T >
+void
+input_stream( T d, fwit& pos )
+{
+  memcpy( &( *pos ), &d, sizeof( d ) );
```

Comment:
I still think this isn't safe for the case where sizeof(d) > sizeof(*pos). I'd decompose into uint_t arrays, and do it one by one --- a good compiler should give you the same thing as a memcpy for the common case